[docs-beta] Refresh Dagster+ Serverless guide (#25563)
dagster-io/dagster: docs/docs-beta/docs/dagster-plus/deployment/serverless/ci-cd-in-serverless.md

---
title: 'CI/CD in Serverless '
displayed_sidebar: 'dagsterPlus'
sidebar_position: 10
---

# CI/CD in Serverless

> If you're a GitHub or GitLab user, our GitHub and GitLab integrations will allow us to set up an automated CI/CD process for you to seamlessly deploy your code locations to Dagster+ and update them whenever you push changes to your project's repo. You can also use other Git providers or a local Git repository with our [dagster-cloud CLI](/dagster-plus/deployment/branch-deployments/dagster-cloud-cli) to run your own CI/CD process.

Reviewer: Have we called these "Integrations" in the past? That term is a little bit overloaded.

```suggestion
If you're a GitHub or GitLab user, you can use our predefined workflows to seamlessly deploy and synchronize your code to Dagster+. You can also use other Git providers or a local Git repository with our [dagster-cloud CLI](/dagster-plus/deployment/branch-deployments/dagster-cloud-cli) to run your own CI/CD process.
```

<Tabs groupId="method">
<TabItem value="GitHub" label="With GitHub">

> If you're a GitHub user, our GitHub integration is the fastest way to get started. It uses a GitHub app and GitHub Actions to set up a repo containing skeleton code and configuration consistent with Dagster+'s best practices with a single click.

Reviewer: This could be simplified if we choose to take the above suggestion ^

- An initial commit will need to be able to be merged directly to the repo's `main` branch to automatically add the GitHub Actions workflow files. If [branch protection rules](https://docs.github.com/en/repositories/configuring-branches-and-merges-in-your-repository/managing-protected-branches/about-protected-branches#about-protected-branches) require changes be done through a pull request, it will prevent the automatic setup from completing.

  - You can temporarily disable the branch protection rules and then re-enable them after the automatic setup completes. Alternatively, you can manually set up the GitHub Actions workflows. You can use our [dagster-cloud-serverless-quickstart repo](https://github.com/dagster-io/dagster-cloud-serverless-quickstart) as a template and its [README](https://github.com/dagster-io/dagster-cloud-serverless-quickstart/blob/main/README.md) as a guide.

:::

</TabItem>

<TabItem value="GitLab" label="With GitLab">

> If you're a GitLab user, our GitLab integration is the fastest way to get started. It uses a GitLab app to set up a repo containing skeleton code and CI/CD configuration consistent with Dagster+'s best practices with a single click.

Reviewer: Same as the feedback for the GitHub section ^
